packages/web/src/common/store/pages/audio-rewards/sagas.ts

```diff
@@ -536,7 +564,12 @@ function* claimChallengeRewardAsync(
   action: ReturnType<typeof claimChallengeReward>
 ) {
   try {
-    yield* call(claimSingleChallengeRewardAsync, action)
+    const result = yield* call(claimSingleChallengeRewardAsync, action)
+    if (result?.aaoErrorCode) {
```

Comment:
this could be nicer like

```
try {
    const result = yield* call(claimSingleChallengeRewardAsync, action)
} catch (e) {
    // do different things based on e
     yield* put(
        claimChallengeRewardFailed({ error: e })
      )
}
```

Reply:
yeah, but see other comment which would have me throw the aao error code

Reply:
oh. was trying to grab screenshots for what im trying to explain, but apparently i can do your suggestion. definitely tried it before submitting my PR but got an error.

Reply:
ahhh. the error was in sdk. so the sdk TS setting is diff from web...

<img width="533" alt="Screenshot 2024-05-14 at 5 10 18 PM" src="https://github.com/AudiusProject/audius-protocol/assets/9600175/797eaeb2-d2b3-4f64-99b0-c1147582fe2f">

Reply:
anyway, going to update the web saga to just throw

Reply:
other issue is that we currently log the error. but, we dont want to log the error code. so would have to update that too

Reply:
lol nvm. lint fails on web too for that issue. leaving as-is if you dont mind

![Uploading Screenshot 2024-05-14 at 5.25.42 PM.png…]()